test/server/cards/01_SOR/upgrades/Traitorous.spec.ts

```diff
@@ -0,0 +1,95 @@
+describe('Traitorous', function() {
+    integration(function(contextRef) {
+        describe('Traitorous\'s ability', function() {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine'],
+                        hand: ['traitorous', 'change-of-heart'],
+                    },
+                    player2: {
+                        groundArena: ['superlaser-technician', 'wampa'],
+                        spaceArena: ['cartel-spacer', 'survivors-gauntlet'],
+                        leader: { card: 'luke-skywalker#faithful-friend', deployed: true },
+                        hand: ['disabling-fang-fighter'],
+                    }
+                });
+            });
+
+            it('allows to take control of a non-leader unit that costs 3 or less', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.traitorous);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.superlaserTechnician, context.cartelSpacer, context.wampa, context.lukeSkywalker, context.survivorsGauntlet]);
+                context.player1.clickCard(context.cartelSpacer);
+
+                expect(context.cartelSpacer).toBeInZone('spaceArena', context.player1);
+
+                context.player2.clickCard(context.disablingFangFighter);
+                context.player2.clickCard(context.traitorous);
+
+                expect(context.cartelSpacer).toBeInZone('spaceArena', context.player2);
+            });
+
+            it('does not allow to take control of a leader unit', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.traitorous);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.superlaserTechnician, context.cartelSpacer, context.wampa, context.lukeSkywalker, context.survivorsGauntlet]);
+                context.player1.clickCard(context.lukeSkywalker);
+
+                expect(context.lukeSkywalker).toBeInZone('groundArena', context.player2);
+            });
+
+            it('does not allow to take control of a non-leader unit that costs more than 3', function () {
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.traitorous);
+                expect(context.player1).toBeAbleToSelectExactly([context.battlefieldMarine, context.superlaserTechnician, context.cartelSpacer, context.wampa, context.lukeSkywalker, context.survivorsGauntlet]);
+                context.player1.clickCard(context.wampa);
+
+                expect(context.wampa).toBeInZone('groundArena', context.player2);
+            });
+
+            it('gives back control of a unit when unattached', function () {
```

Comment:
This is great! One small ask, could you add a few comments narrating the general flow of what's happening in the test?